StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift

```diff
@@ -213,7 +205,84 @@ class EmbeddedPaymentMethodsView: UIView {
+                                   savedPaymentMethodAccessoryType: RowButton.RightAccessoryButton.AccessoryType?) {
+        guard let previousSavedPaymentMethodButton = self.savedPaymentMethodButton,
+              let viewIndex = stackView.arrangedSubviews.firstIndex(of: previousSavedPaymentMethodButton) else {
+            return
+        }
+
+        if let savedPaymentMethod = savedPaymentMethods.first {
+            // Replace saved payment method button at same index
+            let updatedSavedPaymentMethodButton = createSavedPaymentMethodButton(savedPaymentMethod: savedPaymentMethod,
+                                                                                 savedPaymentMethodAccessoryType: savedPaymentMethodAccessoryType)
+            // If the previous item was as selected spm or the user explicitly selected a spm, then auto-select saved pm
+            if selection?.isSaved ?? false || userSelectedPaymentMethod != nil {
+                self.stackView.arrangedSubviews.forEach { view in
+                    (view as? RowButton)?.isSelected = false
+                }
+                updatedSavedPaymentMethodButton.isSelected = true
+
+                // When selection is set, updateMandate() should be called without animation to
+                // avoid animation jank when adding and removing items within the stack view
+                self.animateUpdateMandate = false
+                self.selection = .saved(paymentMethod: savedPaymentMethod)
+            }
+            // Remove old button & insert new button
+            stackView.removeArrangedSubview(previousSavedPaymentMethodButton, animated: false)
+            self.stackView.insertArrangedSubview(updatedSavedPaymentMethodButton, at: viewIndex)
+
+            // Update instance states
+            self.savedPaymentMethods = savedPaymentMethods
+            self.savedPaymentMethodButton = updatedSavedPaymentMethodButton
+        } else {
+            // No more payment methods
+            let separatorIndex = stackView.arrangedSubviews.index(before: viewIndex)
+            stackView.removeArrangedSubview(at: separatorIndex, animated: false)
+            stackView.removeArrangedSubview(previousSavedPaymentMethodButton, animated: false)
+
+            if case .saved(let paymentMethod) = selection {
+                let removedPaymentMethodPreviouslySelected = savedPaymentMethods.first { $0.stripeId == paymentMethod.stripeId }
+                if removedPaymentMethodPreviouslySelected == nil {
+                    // Remove the saved payment method selection if we just removed it
+                    selection = nil
+                }
+            }
+
+            // Update instance states
+            self.savedPaymentMethods = savedPaymentMethods
+            self.savedPaymentMethodButton = nil
+        }
+    }
+
+    func createSavedPaymentMethodButton(savedPaymentMethod: STPPaymentMethod,
```

Comment:
```suggestion
    func makeSavedPaymentMethodButton(savedPaymentMethod: STPPaymentMethod,
```
nit per https://www.swift.org/documentation/api-design-guidelines/  (search "make")